carla/recourse_methods/autoencoder/models.py

```diff
@@ -0,0 +1,115 @@
+import os
+
+from keras import backend as K
+from keras.layers import Dense, Input
+from keras.models import Model, Sequential
+
+
+class Autoencoder:
+    def __init__(
+        self, dim_input, dim_hidden_layer1, dim_hidden_layer2, dim_latent, data_name
+    ):
+        """
+        Defines the structure of the autoencoder
+        :param dim_input: int > 0; number of neurons for this layer (for Adult: 104)
+        :param dim_hidden_layer_1: int > 0, number of neurons for this layer (for Adult: 30)
+        :param dim_hidden_layer_2: int > 0, number of neurons for this layer (for Adult: 15)
+        :param dim_latent: int >0; number of dimensions for the latent code
+        """
+
+        self.dim_input = dim_input
+        self.dim_latent = dim_latent
+        self.dim_hidden_layer1 = dim_hidden_layer1
+        self.dim_hidden_layer2 = dim_hidden_layer2
+        self.data_name = data_name
+
+    def train(self, xtrain, xtest, epochs, batch_size):
+        def loss(y_true, y_pred):
+            """ Negative log likelihood (Bernoulli). """
+
+            # Works if data is normalized between 0 and 1!
+            # Keras.losses.binary_crossentropy gives the mean
+            # Over the last axis. we require the sum
+
+            return K.sum(K.binary_crossentropy(y_true, y_pred), axis=-1)
```

Comment:
maybe pass the loss function as parameter, I think some methods use a different one.

Reply:
I like the idea, I'll add the loss function as an optional parameter to guarantee at least one working default loss function.